spec/models/user_spec.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  let(:user) { build(:user) }
+  let(:user2) { build(:user) }
+  let(:user3) { build(:user) }
+
+  it 'creates valid user' do
+    expect(user.save).to be(true)
+  end
+
+  context 'is not saved because it' do
+    it 'forbides blank username' do
+      user = build(:user, username: '')
+      user.save
+      expect(user.errors[:username]).to include("can't be blank")
+    end
+
+    it 'requires valid email' do
```

Comment:
It isn't necessary for this spec to include tests for `devise` checked attributes. This is unless we included some special formats in our `user.rb` that we're allowing for `email`.